src/malli/transform.cljc

```diff
@@ -82,7 +82,9 @@
   (if (string? x)
     (try
       #?(:clj  (Long/parseLong x)
-         :cljs (let [x' (js/parseInt x 10)]
+         :cljs (let [x' (if (re-find #"\D" (subs x 1))
```

Comment:
I think it would be better to use `js/Number` here, and test the result with `js/Number.isInteger`.

Reply:
Would guess the performance is better but also the number logic stays elsewhere. What do you think.

Reply:
After making the pull. I've tried isFinite to get rid of the re-find. But I've abandoned because it let numbers like '2e2' pass. And it's not on par with clojure parseLong. Number/isInteger suffers from the same problem. Number('2e2') = 200 .